dask_expr/_collection.py

```diff
@@ -2654,7 +2660,9 @@ def from_pandas(data, npartitions=None, sort=True, chunksize=None):
     if not has_parallel_type(data):
         raise TypeError("Input must be a pandas DataFrame or Series.")
 
-    if data.index.isna().any() and not is_any_real_numeric_dtype(data.index):
+    if is_any_real_numeric_dtype and (
```

Comment:
I don't like this pattern, can you create a function in ``_util`` that just returns True/False if is_any_real_numeric_dtype is not available? We should keep those checks out of the actual logic as much as possible

Reply:
Yeah, sounds good. Thanks for being open to that. I appreciate your help with this.

Reply:
As long as we can keep the number of these things very very low, dask/dask ci is a pain at the moment because a different upstream version might screw you over on every new pr